File: Back/Forum/src/main/java/com/example/forum/Entity/Comment.java
package com.example.forum.Entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.Set;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table
@Builder
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long idComment ;
    private int likesCount;
    private int dislikesCount;
    private boolean mostPertinentComment;
    private String textComment;
    @Temporal(TemporalType.DATE)
    private Date dateCreationComment;
    private int userId;


    @JsonIgnore
    @ManyToOne
    Post post;

    @OneToMany(cascade = CascadeType.ALL)
    private Set<Dislike> likes;
}

File: Front/src/app/models/comment.ts
import {Post} from './post';
import { User } from './user';

export class Comment{
  idComment!: number;
  likesCount!: number;
  dislikesCount!:number;
  mostPertinentComment!: boolean;
  textComment!: string;
  dateCreationComment!: string ;
  post!: Post;
  userId:number
  fullUser:User
  ActiveLike:Like;
  likes!: Like[];
}
export class Like {
  id: number;
  userID: number;
  eventID: number;
  commentID: number;
  status: string;
  datetime: Date;
}